src/tests/Microsoft.PowerFx.Core.Tests.Shared/AnalyzeThreadSafety.cs

```diff
@@ -192,22 +189,65 @@ public static void CheckStatics(Assembly asm, HashSet<Type> bugsFieldType, HashS
             Assert.Empty(errors);
         }
 
-        // $$$ Supersedes ImmutabilityTests.
-        // This is more aggressive (includes private fields), but they don't all pass. So assert is disabled.
-        public static void CheckImmutableTypes(Assembly[] assemblies, bool enableAssert = false)
+        // Does the property have 'init' keyword.
+        private static bool HasInitKeyword(PropertyInfo prop)
+        {
+            var attrs = prop.SetMethod.ReturnParameter.GetRequiredCustomModifiers();
+
+            // Can't use typeof() because it may not be available in .Net 5.0+
+            // So check type name instead. 
+            // typeof(System.Runtime.CompilerServices.IsExternalInit)
+            bool hasInit = attrs.Any(type => type.Name == "IsExternalInit");
```

Comment:
This was key fix. Before, we used typeof() , which mandated ifdefs and didn't work on all platforms. 
Just compare the type name instead.